src/Task.ts

```diff
@@ -371,16 +443,49 @@ export function getTokensSource(address: Address): Address {
   return Address.zero()
 }
 
-export function loadOrCreateTaskConfig(taskId: string): TaskConfig {
-  let taskConfig = TaskConfig.load(taskId)
+export function loadOrCreateTaskConfig(task: Task, block: ethereum.Block): TaskConfig {
+  const taskConfigId = task.taskConfig
+
+  let taskConfig = TaskConfig.load(taskConfigId!)
 
   if (taskConfig === null) {
+    const taskId = task.id + 'block' + block.number.toHexString()
     taskConfig = new TaskConfig(taskId)
     taskConfig.task = taskId
     taskConfig.acceptanceList = loadOrCreateAcceptanceList(taskId).id
     taskConfig.previousBalanceConnector = '0x0000000000000000000000000000000000000000000000000000000000000000'
     taskConfig.nextBalanceConnector = '0x0000000000000000000000000000000000000000000000000000000000000000'
     taskConfig.save()
+  } else {
+    const taskId = task.id + 'block' + block.number.toHexString()
```

Comment:
This can be computed only once out of the `if` context:

```suggestion
    const taskId = task.id + '#' + block.number.toString()
```